### ISSUE TITLE
YACC should apply the exclusions first before applying any rule filters.

### ISSUE THREAD

**Issue by wetarded1** [mohamicorp/yet-another-commit-checker#250]
I am trying to run a case where I have a branch exclusion for a specific feature branch prefix, basically '^feature\/SIP'.  I also have a JIRA query search enabled for the repository as well.  The problem is I have a user that is playing in the feature/SIP area that does not have access to the projects in the JIRA filter (and cant have access to them).  So when this user attempts to push, the JIRA query fails when in fact, the query should not be getting executed at all because the branch is in the excluded branches.

**Comment by george-vulov** [comment]
As far as I can tell, all Jira queries are done after the branch exclusion pattern is resolved. Can you enable debug logging, reproduce the failed push, and post the output here?

Enable debugging like this, except replace the username and instance URL:
curl -u admin -v -X PUT -d "" -H "Content-Type: application/json" http://localhost:7990/bitbucket/rest/api/latest/logs/logger/com.isroot/debug

**Comment by wetarded1** [comment]
I unfortunately do not have access to our bitbucket server at that level.  Its managed by higher powers.  When we ran it on our end the output from the commit was the output we put in place in the tool for when a JIRA commit was rejected.  To reproduce this, setup a JIRA filter that either doesnt exist or you dont have access to and an exclusion rule for the branch in question.  Then try to push.  I will try to work a bit more at it and see if I can get it to happen and document exactly what I did to make it happen.

**Comment by george-vulov** [comment]
Alright, if you can reproduce the issue again, please capture this information and attach it here:
* Screenshots of the YACC settings for the repository in question. Scroll down to make sure you capture all the settings.
* Include the full output you see what you try to push (including the branch name)

Thanks!

**Comment by wetarded1** [comment]
I am going to close this out.  It was a misunderstanding of what the exclusion filter was looking for.  I gave it a regex that would match the first part of the feature branch subsection (^feature\/SIP) which would normally do a partial match against something like  feature/SIP-ABCD but the tool wants a full match so I changed it around to wildcard the end of the regex and now it appears to be working.
